### PR TITLE
client: Added a Dockerimage

### PR DESCRIPTION
**Pull request by Bullrich**
Added a docker image for the client, in the case that we want to deploy the site to any other platform aside from GitHub Pages

### CODE REVIEW

**Review by rzadp — APPROVED** [review]

File: client/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+WORKDIR /app
+
+COPY package.json yarn.lock ./
+RUN yarn install --frozen-lockfile
+COPY . ./
+
+ARG DEMO
+ENV VITE_DEMO ${DEMO}
+ARG CAPTCHA_KEY
```
Comment (PierreBesson) [inline client/Dockerfile:11]: Those variables should not be set in docker build args but by docker environment variables. If those environment specific variables are set in the docker image itself, the image will not longer be portable to multiple environments without rebuilding.

**Review by PierreBesson — CHANGES_REQUESTED** [review]
There should be a CI job that pushes this image to `paritytech/faucet-ui` with the same tags as the backend.

**Review by PierreBesson — DISMISSED** [review]

**Comment by Bullrich** [comment]
Hi @PierreBesson! I added the logic to build the image and publish it in `gitlab-ci`.

Could you please review that file and let me know if it is as intended? I'm afraid my knowledge of gitlab's deployment script is not very broad

**Comment by rzadp** [comment]
Should we update `publish-docker-image-description` as well?

**Review by mordamax — APPROVED** [review]

**Comment by sergejparity** [comment]
> Should we update `publish-docker-image-description` as well?

Yes, this make sense. I'll add the commit for that.

**Review by sergejparity — APPROVED** [review]

**Pull request merged**